src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -138,17 +138,21 @@ const ChatsList: React.FunctionComponent = () => {
         <div className="private-chat-user">
           Chatting with: {selectedUser.partyName}
         </div>
-        <ChatBox chats={chatsToDisplay} onMessageSubmit={submitMessage} />
+        <ChatBox
+          usersById={chatUsers}
+          chats={chatsToDisplay}
+          onMessageSubmit={submitMessage}
+        />
       </Fragment>
     );
   }
 
   return (
     <Fragment>
+      <UserSearchBar onSelect={setSelectedUser} />
       {hasPrivateChats && (
         <div className="private-container show">
           <div className="private-messages-list">
-            <UserSearchBar onSelect={setSelectedUser} />
             {discussions.map((userId: string) => {
               const sender = { ...chatUsers![userId], id: userId };
               const lastMessageExchanged =
```

Comment:
[OOS, future] This probably should be extracted + `useMemo`, or reworked as a sub component (or both)